onnxruntime/core/session/inference_session.cc

```diff
@@ -2245,21 +2262,30 @@ void InferenceSession::InitLogger(logging::LoggingManager* logging_manager) {
 common::Status InferenceSession::AddPredefinedTransformers(GraphTransformerManager& transformer_manager,
                                                            TransformerLevel graph_optimization_level,
                                                            bool saving_runtime_optimizations) const {
+#if !defined(ORT_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION)
+  ORT_RETURN_IF(saving_runtime_optimizations, "Saving runtime optimizations is not supported in this build.");
+#endif  // !defined(ORT_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION)
+
```

Comment:
I would have expected ORT_ENABLE_ORT_FORMAT_RUNTIME_GRAPH_OPTIMIZATION to only be applied to a minimal build to add the code for applying the saved optimizations. 

In a full build you should always be able to save the optimizations, with that being turned on via config setting.

Reply:
I took the approach of having the general runtime optimization functionality be optional. We could have it always be enabled in a full build.

I guess we could also do something like this:
`#if !defined(ORT_MINIMAL_BUILD) || (defined(ORT_EXTENDED_MINIMAL_BUILD) && defined(ORT_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD))`

Not sure that's better though.

Reply:
I'm not sure of a scenario where you don't want it to be included in a full build. We'd certainly need it enabled in any pre-built package so that can be used to create the saved optimizations. 

Does ORT_ENABLE_RUNTIME_OPTIMIZATION_IN_MINIMAL_BUILD imply ORT_EXTENDED_MINIMAL_BUILD so things can be simplified? 

How many places would we need to use this check though? AddPredefinedTransformers is not enabled at all in a minimal build currently.